horton/test/common.py

```diff
@@ -372,3 +376,32 @@ def numpy_seed(seed=1):
     np.random.seed(seed)
     yield None
     np.random.set_state(state)
+
+
+@contextmanager
+def truncated_file(name, fn_orig, nline, nadd):
+    """Make a temporary truncated copy of a file.
+
+    Parameters
+    ----------
+    name : str
+           The name of test, used to make a unique temporary directory
+    fn_orig : str
+              The file to be truncated.
+    nline : int
+            The number of lines to retain.
+    nadd : int
+           The number of empty lines to add.
+    """
+    with tmpdir(name) as dn:
+        fn_truncated = '%s/truncated_%i_%s' % (dn, nline, os.path.basename(fn_orig))
+        with open(fn_orig) as f_orig, open(fn_truncated, 'w') as f_truncated:
+            counter = 0
```

Comment:
I think this can be more pythonic if you use an enumerate instead of the explicit counter?